python/mlx/nn/layers/transformer.py

```diff
@@ -176,10 +179,11 @@ def __init__(
             for i in range(num_layers)
```

Comment:
You could do the checkpointing here based on the argument using `nn.checkpoint`?